calm_adapter/window_generator/window_generator.py

```diff
@@ -0,0 +1,71 @@
+#!/bin/env python3
+
+from datetime import date, timedelta
+import click
+import json
+import boto3
+from tqdm import tqdm
+
+
+class WindowGenerator:
+
+    topic_arn = "arn:aws:sns:eu-west-1:760097843905:calm-adapter"
+    role_arn = "arn:aws:iam::760097843905:role/platform-developer"
+
+    def __init__(self, start, end=None):
+        self.start = start
+        self.end = end or start
+        assert self.start <= self.end, "Start window not be after end window"
+        self.client = self.sns_client
+
+    def run(self):
+        for window in tqdm([*self.windows]):
+            self.send_window_to_sns(window)
+
+    @property
+    def windows(self):
+        while self.start <= self.end:
+            yield self.start
+            self.start += timedelta(days=1)
+
+    def send_window_to_sns(self, window):
+        msg = json.dumps({"date": window.isoformat()})
+        self.client.publish(
+            TopicArn=self.topic_arn, Message=msg, Subject=f"Window sent by {__file__}"
+        )
+
+    @property
+    def sns_client(self):
+        sts = boto3.client("sts")
+        role = sts.assume_role(
+            RoleArn=self.role_arn, RoleSessionName="AssumeRoleSession1"
+        )
+        credentials = role["Credentials"]
+        return boto3.client(
+            "sns",
+            aws_access_key_id=credentials["AccessKeyId"],
+            aws_secret_access_key=credentials["SecretAccessKey"],
+            aws_session_token=credentials["SessionToken"],
+        )
+
+
+@click.group()
+def cli():
+    pass
+
+
+@cli.command()
+def full_ingest():
+    start = date(2015, 9, 30)
```

Comment:
What's the reason for this particular start date?

Reply:
That's what I found to be the first date where any records are returned. Can add a comment saying that much.